src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -120,10 +127,14 @@ export class FileService implements IFileService {
 		}
 
 		if (this.basePath && !this.options.disableWatcher) {
-			if (isWindows) {
-				this.setupWin32WorkspaceWatching();
+			if (this.options.useNsfwFileWatcher) {
```

Comment:
@bpasero merging this now as the tests are passing and the new code should all be blocked off by this setting.